File: algorithm-leetcode-python/greedy/LeetCode1702.py
class LeetCode1702(object):
    def maximumBinaryString(self, binary: str) -> str:
        index = binary.find('0')
        if index < 0:
            return binary
        count = binary.count('1', index)
        return '1' * (len(binary) - count - 1) + '0' + '1' * count

File: algorithm-leetcode-jvm/src/main/kotlin/org/muxun/algorithm/leetcode/kotlin/greedy/LeetCode1702.kt
package org.muxun.algorithm.leetcode.kotlin.greedy

class LeetCode1702 {

    fun maximumBinaryString(binary: String): String {
        var index = binary.indexOf('0')
        if (index < 0) {
            return binary
        }
        val n = binary.length
        val arr = binary.toCharArray()
        var count = 0
        index++
        while (index < n) {
            count += arr[index++] - '0'
        }
        return "1".repeat(n - count - 1) + "0" + "1".repeat(count)
    }
}